app/controllers/application_controller.rb

```diff
@@ -1,7 +1,22 @@
+
 class ApplicationController < ActionController::Base
   # Prevent CSRF attacks by raising an exception.
   # For APIs, you may want to use :null_session instead.
-  protect_from_forgery with: :exception
+  protect_from_forgery with: :null_session
+  skip_before_filter :verify_authenticity_token, :only => :create
+  before_action :current_user
+
+  def current_user
+    if session[:user_id] && !!User.exists?(session[:user_id])
```

Comment:
Okay, my Ruby might just be rusty but it looks like this is repetitive. The `!!` is useful for returning a boolean, particularly when you're not expecting one… but `.exists?`[ _always_ returns a boolean](http://apidock.com/rails/ActiveRecord/FinderMethods/exists%3F). Also looks like `User(session[:user_id]).exists?` should be possible, which would be more legible IMO.